sdk/compiler/damlc/lib/DA/Cli/Options.hs

```diff
@@ -573,14 +574,32 @@ optionsParser numProcessors enableScenarioService parsePkgName parseDlintUsage =
         <> help "Set path to CPP."
         <> internal
 
-warnBadInterfaceInstancesOpt :: Parser WarnBadInterfaceInstances
-warnBadInterfaceInstancesOpt =
-  WarnBadInterfaceInstances <$>
-  flagYesNoAuto
-    "warn-bad-interface-instances"
-    False
-    "Convert errors about bad, non-upgradeable interface instances into warnings."
-    idm
+    optUpgradeDar :: Parser (Maybe FilePath)
+    optUpgradeDar = optional . optionOnce str
+        $ metavar "UPGRADE_DAR"
+        <> long "upgrades"
+        <> help "Set DAR to upgrade"
+
+    optTypecheckUpgrades :: Parser YesNoAuto
+    optTypecheckUpgrades =
+      flagYesNoAuto'
+        "typecheck-upgrades"
```

Comment:
We've moved the `typecheck-upgrades:` daml.yaml field into a build option - this is fine because we never documented the daml.yaml field and it's only been out very briefly.

Users that want to disable typechecking using a compiler that only supports the field can comment out the `upgrades:` field and enable --warn-bad-interface-instances for remaining interface errors (if any).